pandas/io/excel.py

```diff
@@ -156,47 +340,22 @@ def read_excel(io, sheetname=0, **kwds):
 class ExcelFile(object):
     """
     Class for parsing tabular excel sheets into DataFrame objects.
-    Uses xlrd. See ExcelFile.parse for more documentation
+    Uses xlrd and/or ezodf. See ExcelFile.parse for more documentation
 
     Parameters
     ----------
-    io : string, file-like object or xlrd workbook
-        If a string, expected to be a path to xls or xlsx file
+    io : string, file-like object or xlrd/ezodf workbook
+        If a string, expected to be a path to xls, xlsx, or ods file
     engine: string, default None
         If io is not a buffer or path, this must be set to identify io.
-        Acceptable values are None or xlrd
+        Acceptable values are None, xlrd, or ezodf
     """
-    def __init__(self, io, **kwds):
-
-        import xlrd  # throw an ImportError if we need to
-
-        ver = tuple(map(int, xlrd.__VERSION__.split(".")[:2]))
-        if ver < (0, 9):  # pragma: no cover
-            raise ImportError("pandas requires xlrd >= 0.9.0 for excel "
-                              "support, current version " + xlrd.__VERSION__)
 
+    def __init__(self, io, **kwds):
         self.io = io
 
         engine = kwds.pop('engine', None)
-
-        if engine is not None and engine != 'xlrd':
-            raise ValueError("Unknown engine: %s" % engine)
-
-        if isinstance(io, compat.string_types):
-            if _is_url(io):
-                data = _urlopen(io).read()
-                self.book = xlrd.open_workbook(file_contents=data)
-            else:
-                self.book = xlrd.open_workbook(io)
-        elif engine == 'xlrd' and isinstance(io, xlrd.Book):
-            self.book = io
-        elif not isinstance(io, xlrd.Book) and hasattr(io, "read"):
-            # N.B. xlrd.Book has a read attribute too
-            data = io.read()
-            self.book = xlrd.open_workbook(file_contents=data)
-        else:
-            raise ValueError('Must explicitly set engine if not passing in'
-                             ' buffer or path for io.')
+        self.engine, self.book = BaseFile().create(io, engine=engine)
```

Comment:
you want to save the file object

e.g.

```
self.factory = BaseFile.create(io, engine=engine)
```

``.engine, .book`` are just properties on the factory